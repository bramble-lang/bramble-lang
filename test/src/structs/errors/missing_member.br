fn my_main() -> i64 {
    let t: MyStruct := MyStruct{
        x: 5, 
        y: 1, 
    };
    root::std::io::writei64ln(t.x);
    root::std::io::writei64ln(t.y);
    root::std::io::writeboolln(t.z);
    
    return 0;
}

struct MyStruct {
    x: i64,
    y: i64,
    z: bool,
}