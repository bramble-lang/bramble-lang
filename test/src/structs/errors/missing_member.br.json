[{"id": 1, "stage": "lexer", "source": [0, 2], "ok": "Keyword"},
{"id": 2, "stage": "lexer", "source": [3, 10], "ok": "Identifier"},
{"id": 3, "stage": "lexer", "source": [10, 11], "ok": "Operator"},
{"id": 4, "stage": "lexer", "source": [11, 12], "ok": "Operator"},
{"id": 5, "stage": "lexer", "source": [13, 15], "ok": "Operator"},
{"id": 6, "stage": "lexer", "source": [16, 19], "ok": "Primitive"},
{"id": 7, "stage": "lexer", "source": [20, 21], "ok": "Operator"},
{"id": 8, "stage": "lexer", "source": [26, 29], "ok": "Keyword"},
{"id": 9, "stage": "lexer", "source": [30, 31], "ok": "Identifier"},
{"id": 10, "stage": "lexer", "source": [31, 32], "ok": "Operator"},
{"id": 11, "stage": "lexer", "source": [33, 41], "ok": "Identifier"},
{"id": 12, "stage": "lexer", "source": [42, 44], "ok": "Operator"},
{"id": 13, "stage": "lexer", "source": [45, 53], "ok": "Identifier"},
{"id": 14, "stage": "lexer", "source": [53, 54], "ok": "Operator"},
{"id": 15, "stage": "lexer", "source": [63, 64], "ok": "Identifier"},
{"id": 16, "stage": "lexer", "source": [64, 65], "ok": "Operator"},
{"id": 17, "stage": "lexer", "source": [66, 67], "ok": "Integer"},
{"id": 18, "stage": "lexer", "source": [67, 68], "ok": "Operator"},
{"id": 19, "stage": "lexer", "source": [78, 79], "ok": "Identifier"},
{"id": 20, "stage": "lexer", "source": [79, 80], "ok": "Operator"},
{"id": 21, "stage": "lexer", "source": [81, 82], "ok": "Integer"},
{"id": 22, "stage": "lexer", "source": [82, 83], "ok": "Operator"},
{"id": 23, "stage": "lexer", "source": [89, 90], "ok": "Operator"},
{"id": 24, "stage": "lexer", "source": [90, 91], "ok": "Operator"},
{"id": 25, "stage": "lexer", "source": [96, 103], "ok": "Keyword"},
{"id": 26, "stage": "lexer", "source": [103, 105], "ok": "Operator"},
{"id": 27, "stage": "lexer", "source": [105, 108], "ok": "Identifier"},
{"id": 28, "stage": "lexer", "source": [108, 110], "ok": "Operator"},
{"id": 29, "stage": "lexer", "source": [110, 112], "ok": "Identifier"},
{"id": 30, "stage": "lexer", "source": [112, 114], "ok": "Operator"},
{"id": 31, "stage": "lexer", "source": [114, 124], "ok": "Identifier"},
{"id": 32, "stage": "lexer", "source": [124, 125], "ok": "Operator"},
{"id": 33, "stage": "lexer", "source": [125, 126], "ok": "Identifier"},
{"id": 34, "stage": "lexer", "source": [126, 127], "ok": "Operator"},
{"id": 35, "stage": "lexer", "source": [127, 128], "ok": "Identifier"},
{"id": 36, "stage": "lexer", "source": [128, 129], "ok": "Operator"},
{"id": 37, "stage": "lexer", "source": [129, 130], "ok": "Operator"},
{"id": 38, "stage": "lexer", "source": [135, 142], "ok": "Keyword"},
{"id": 39, "stage": "lexer", "source": [142, 144], "ok": "Operator"},
{"id": 40, "stage": "lexer", "source": [144, 147], "ok": "Identifier"},
{"id": 41, "stage": "lexer", "source": [147, 149], "ok": "Operator"},
{"id": 42, "stage": "lexer", "source": [149, 151], "ok": "Identifier"},
{"id": 43, "stage": "lexer", "source": [151, 153], "ok": "Operator"},
{"id": 44, "stage": "lexer", "source": [153, 163], "ok": "Identifier"},
{"id": 45, "stage": "lexer", "source": [163, 164], "ok": "Operator"},
{"id": 46, "stage": "lexer", "source": [164, 165], "ok": "Identifier"},
{"id": 47, "stage": "lexer", "source": [165, 166], "ok": "Operator"},
{"id": 48, "stage": "lexer", "source": [166, 167], "ok": "Identifier"},
{"id": 49, "stage": "lexer", "source": [167, 168], "ok": "Operator"},
{"id": 50, "stage": "lexer", "source": [168, 169], "ok": "Operator"},
{"id": 51, "stage": "lexer", "source": [174, 181], "ok": "Keyword"},
{"id": 52, "stage": "lexer", "source": [181, 183], "ok": "Operator"},
{"id": 53, "stage": "lexer", "source": [183, 186], "ok": "Identifier"},
{"id": 54, "stage": "lexer", "source": [186, 188], "ok": "Operator"},
{"id": 55, "stage": "lexer", "source": [188, 190], "ok": "Identifier"},
{"id": 56, "stage": "lexer", "source": [190, 192], "ok": "Operator"},
{"id": 57, "stage": "lexer", "source": [192, 203], "ok": "Identifier"},
{"id": 58, "stage": "lexer", "source": [203, 204], "ok": "Operator"},
{"id": 59, "stage": "lexer", "source": [204, 205], "ok": "Identifier"},
{"id": 60, "stage": "lexer", "source": [205, 206], "ok": "Operator"},
{"id": 61, "stage": "lexer", "source": [206, 207], "ok": "Identifier"},
{"id": 62, "stage": "lexer", "source": [207, 208], "ok": "Operator"},
{"id": 63, "stage": "lexer", "source": [208, 209], "ok": "Operator"},
{"id": 64, "stage": "lexer", "source": [219, 225], "ok": "Keyword"},
{"id": 65, "stage": "lexer", "source": [226, 227], "ok": "Integer"},
{"id": 66, "stage": "lexer", "source": [227, 228], "ok": "Operator"},
{"id": 67, "stage": "lexer", "source": [229, 230], "ok": "Operator"},
{"id": 68, "stage": "lexer", "source": [232, 238], "ok": "Keyword"},
{"id": 69, "stage": "lexer", "source": [239, 247], "ok": "Identifier"},
{"id": 70, "stage": "lexer", "source": [248, 249], "ok": "Operator"},
{"id": 71, "stage": "lexer", "source": [254, 255], "ok": "Identifier"},
{"id": 72, "stage": "lexer", "source": [255, 256], "ok": "Operator"},
{"id": 73, "stage": "lexer", "source": [257, 260], "ok": "Primitive"},
{"id": 74, "stage": "lexer", "source": [260, 261], "ok": "Operator"},
{"id": 75, "stage": "lexer", "source": [266, 267], "ok": "Identifier"},
{"id": 76, "stage": "lexer", "source": [267, 268], "ok": "Operator"},
{"id": 77, "stage": "lexer", "source": [269, 272], "ok": "Primitive"},
{"id": 78, "stage": "lexer", "source": [272, 273], "ok": "Operator"},
{"id": 79, "stage": "lexer", "source": [278, 279], "ok": "Identifier"},
{"id": 80, "stage": "lexer", "source": [279, 280], "ok": "Operator"},
{"id": 81, "stage": "lexer", "source": [281, 285], "ok": "Primitive"},
{"id": 82, "stage": "lexer", "source": [285, 286], "ok": "Operator"},
{"id": 83, "stage": "lexer", "source": [287, 288], "ok": "Operator"},
{"id": 89, "parent_id": 87, "stage": "parser", "source": [16, 19], "ok": "Primitive Type"},
{"id": 87, "parent_id": 86, "stage": "parser", "source": [0, 19], "ok": "Routine Declaration"},
{"id": 94, "parent_id": 93, "stage": "parser", "source": [33, 41], "ok": "Path"},
{"id": 93, "parent_id": 92, "stage": "parser", "source": [33, 41], "ok": "Custom Type"},
{"id": 92, "parent_id": 91, "stage": "parser", "source": [30, 41], "ok": "Identifier Declaration"},
{"id": 104, "parent_id": 103, "stage": "parser", "source": [45, 53], "ok": "Path"},
{"id": 115, "parent_id": 111, "stage": "parser", "source": [66, 67], "ok": "Number"},
{"id": 111, "parent_id": 110, "stage": "parser", "source": [66, 67]},
{"id": 110, "parent_id": 109, "stage": "parser", "source": [66, 67]},
{"id": 109, "parent_id": 108, "stage": "parser", "source": [66, 67]},
{"id": 108, "parent_id": 107, "stage": "parser", "source": [66, 67]},
{"id": 107, "parent_id": 106, "stage": "parser", "source": [66, 67]},
{"id": 106, "parent_id": 105, "stage": "parser", "source": [66, 67]},
{"id": 125, "parent_id": 121, "stage": "parser", "source": [81, 82], "ok": "Number"},
{"id": 121, "parent_id": 120, "stage": "parser", "source": [81, 82]},
{"id": 120, "parent_id": 119, "stage": "parser", "source": [81, 82]},
{"id": 119, "parent_id": 118, "stage": "parser", "source": [81, 82]},
{"id": 118, "parent_id": 117, "stage": "parser", "source": [81, 82]},
{"id": 117, "parent_id": 116, "stage": "parser", "source": [81, 82]},
{"id": 116, "parent_id": 105, "stage": "parser", "source": [81, 82]},
{"id": 105, "parent_id": 103, "stage": "parser", "source": [53, 90], "ok": "Struct Exprssion Parameters"},
{"id": 103, "parent_id": 101, "stage": "parser", "source": [45, 90], "ok": "Struct Expression"},
{"id": 101, "parent_id": 100, "stage": "parser", "source": [45, 90]},
{"id": 100, "parent_id": 99, "stage": "parser", "source": [45, 90]},
{"id": 99, "parent_id": 98, "stage": "parser", "source": [45, 90]},
{"id": 98, "parent_id": 97, "stage": "parser", "source": [45, 90]},
{"id": 97, "parent_id": 96, "stage": "parser", "source": [45, 90]},
{"id": 96, "parent_id": 91, "stage": "parser", "source": [45, 90]},
{"id": 91, "parent_id": 90, "stage": "parser", "source": [26, 90], "ok": "Let Binding"},
{"id": 90, "parent_id": 86, "stage": "parser", "source": [26, 91], "ok": "Statement Bind"},
{"id": 137, "parent_id": 136, "stage": "parser", "source": [96, 124], "ok": "Path"},
{"id": 146, "parent_id": 145, "stage": "parser", "source": [125, 126], "ok": "Path"},
{"id": 145, "parent_id": 143, "stage": "parser", "source": [125, 126], "ok": "Identifier"},
{"id": 148, "parent_id": 143, "stage": "parser", "source": [127, 128], "ok": "Structure Field"},
{"id": 143, "parent_id": 142, "stage": "parser", "source": [125, 128], "ok": "Subdata Access: Member Access"},
{"id": 142, "parent_id": 141, "stage": "parser", "source": [125, 128]},
{"id": 141, "parent_id": 140, "stage": "parser", "source": [125, 128]},
{"id": 140, "parent_id": 139, "stage": "parser", "source": [125, 128]},
{"id": 139, "parent_id": 138, "stage": "parser", "source": [125, 128]},
{"id": 138, "parent_id": 136, "stage": "parser", "source": [125, 128]},
{"id": 136, "parent_id": 134, "stage": "parser", "source": [96, 129], "ok": "Routine Call"},
{"id": 134, "parent_id": 133, "stage": "parser", "source": [96, 129]},
{"id": 133, "parent_id": 132, "stage": "parser", "source": [96, 129]},
{"id": 132, "parent_id": 131, "stage": "parser", "source": [96, 129]},
{"id": 131, "parent_id": 130, "stage": "parser", "source": [96, 129]},
{"id": 130, "parent_id": 129, "stage": "parser", "source": [96, 129]},
{"id": 129, "parent_id": 126, "stage": "parser", "source": [96, 129]},
{"id": 126, "parent_id": 86, "stage": "parser", "source": [96, 130], "ok": "Statement Expression"},
{"id": 160, "parent_id": 159, "stage": "parser", "source": [135, 163], "ok": "Path"},
{"id": 169, "parent_id": 168, "stage": "parser", "source": [164, 165], "ok": "Path"},
{"id": 168, "parent_id": 166, "stage": "parser", "source": [164, 165], "ok": "Identifier"},
{"id": 171, "parent_id": 166, "stage": "parser", "source": [166, 167], "ok": "Structure Field"},
{"id": 166, "parent_id": 165, "stage": "parser", "source": [164, 167], "ok": "Subdata Access: Member Access"},
{"id": 165, "parent_id": 164, "stage": "parser", "source": [164, 167]},
{"id": 164, "parent_id": 163, "stage": "parser", "source": [164, 167]},
{"id": 163, "parent_id": 162, "stage": "parser", "source": [164, 167]},
{"id": 162, "parent_id": 161, "stage": "parser", "source": [164, 167]},
{"id": 161, "parent_id": 159, "stage": "parser", "source": [164, 167]},
{"id": 159, "parent_id": 157, "stage": "parser", "source": [135, 168], "ok": "Routine Call"},
{"id": 157, "parent_id": 156, "stage": "parser", "source": [135, 168]},
{"id": 156, "parent_id": 155, "stage": "parser", "source": [135, 168]},
{"id": 155, "parent_id": 154, "stage": "parser", "source": [135, 168]},
{"id": 154, "parent_id": 153, "stage": "parser", "source": [135, 168]},
{"id": 153, "parent_id": 152, "stage": "parser", "source": [135, 168]},
{"id": 152, "parent_id": 149, "stage": "parser", "source": [135, 168]},
{"id": 149, "parent_id": 86, "stage": "parser", "source": [135, 169], "ok": "Statement Expression"},
{"id": 183, "parent_id": 182, "stage": "parser", "source": [174, 203], "ok": "Path"},
{"id": 192, "parent_id": 191, "stage": "parser", "source": [204, 205], "ok": "Path"},
{"id": 191, "parent_id": 189, "stage": "parser", "source": [204, 205], "ok": "Identifier"},
{"id": 194, "parent_id": 189, "stage": "parser", "source": [206, 207], "ok": "Structure Field"},
{"id": 189, "parent_id": 188, "stage": "parser", "source": [204, 207], "ok": "Subdata Access: Member Access"},
{"id": 188, "parent_id": 187, "stage": "parser", "source": [204, 207]},
{"id": 187, "parent_id": 186, "stage": "parser", "source": [204, 207]},
{"id": 186, "parent_id": 185, "stage": "parser", "source": [204, 207]},
{"id": 185, "parent_id": 184, "stage": "parser", "source": [204, 207]},
{"id": 184, "parent_id": 182, "stage": "parser", "source": [204, 207]},
{"id": 182, "parent_id": 180, "stage": "parser", "source": [174, 208], "ok": "Routine Call"},
{"id": 180, "parent_id": 179, "stage": "parser", "source": [174, 208]},
{"id": 179, "parent_id": 178, "stage": "parser", "source": [174, 208]},
{"id": 178, "parent_id": 177, "stage": "parser", "source": [174, 208]},
{"id": 177, "parent_id": 176, "stage": "parser", "source": [174, 208]},
{"id": 176, "parent_id": 175, "stage": "parser", "source": [174, 208]},
{"id": 175, "parent_id": 172, "stage": "parser", "source": [174, 208]},
{"id": 172, "parent_id": 86, "stage": "parser", "source": [174, 209], "ok": "Statement Expression"},
{"id": 220, "parent_id": 216, "stage": "parser", "source": [226, 227], "ok": "Number"},
{"id": 216, "parent_id": 215, "stage": "parser", "source": [226, 227]},
{"id": 215, "parent_id": 214, "stage": "parser", "source": [226, 227]},
{"id": 214, "parent_id": 213, "stage": "parser", "source": [226, 227]},
{"id": 213, "parent_id": 212, "stage": "parser", "source": [226, 227]},
{"id": 212, "parent_id": 211, "stage": "parser", "source": [226, 227]},
{"id": 211, "parent_id": 210, "stage": "parser", "source": [226, 227]},
{"id": 210, "parent_id": 86, "stage": "parser", "source": [219, 227], "ok": "Return"},
{"id": 86, "parent_id": 84, "stage": "parser", "source": [0, 230], "ok": "Funtion Definition"},
{"id": 223, "parent_id": 222, "stage": "parser", "source": [257, 260], "ok": "Primitive Type"},
{"id": 222, "parent_id": 221, "stage": "parser", "source": [254, 260], "ok": "Identifier Declaration"},
{"id": 225, "parent_id": 224, "stage": "parser", "source": [269, 272], "ok": "Primitive Type"},
{"id": 224, "parent_id": 221, "stage": "parser", "source": [266, 272], "ok": "Identifier Declaration"},
{"id": 227, "parent_id": 226, "stage": "parser", "source": [281, 285], "ok": "Primitive Type"},
{"id": 226, "parent_id": 221, "stage": "parser", "source": [278, 285], "ok": "Identifier Declaration"},
{"id": 221, "parent_id": 84, "stage": "parser", "source": [232, 288], "ok": "Struct Definition"},
{"id": 84, "stage": "parser", "source": [0, 288], "ok": "File Module"},
{"id": 230, "stage": "canonize-item-path", "source": [0, 288], "ok": "$missing_member::missing_member"},
{"id": 231, "stage": "canonize-item-path", "source": [0, 288], "ok": "$missing_member::missing_member::missing_member"},
{"id": 232, "stage": "canonize-item-path", "source": [0, 230], "ok": "$missing_member::missing_member::my_main"},
{"id": 233, "stage": "canonize-item-path", "source": [232, 288], "ok": "$missing_member::missing_member::MyStruct"},
{"id": 234, "stage": "canonize-item-path", "source": [254, 260], "ok": "$missing_member::missing_member::x"},
{"id": 235, "stage": "canonize-item-path", "source": [266, 272], "ok": "$missing_member::missing_member::y"},
{"id": 236, "stage": "canonize-item-path", "source": [278, 285], "ok": "$missing_member::missing_member::z"},
{"id": 237, "stage": "canonize-type-ref", "source": [26, 91], "ok": "$missing_member::missing_member::MyStruct"},
{"id": 238, "stage": "canonize-type-ref", "source": [45, 90], "ok": "$missing_member::missing_member::MyStruct"},
{"id": 239, "stage": "type-resolver", "source": [45, 90], "error": "L2-5: Struct expected 3 parameters but found 2"}]