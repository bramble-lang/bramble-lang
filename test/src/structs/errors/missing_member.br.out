Error: Semantic: L2: expected 3 parameters but found 2
